### PR TITLE
force sticky-ad margin-bottom to be 0

### PR DESCRIPTION
**Pull request by zhouyx**
if user add margin to sticky-ad, it will add gap between sticky-ad and bottom of the page. need to  add `margin-bottom: 0 !important` to force it to be at the bottom of the page?

### CODE REVIEW

File: extensions/amp-sticky-ad/0.1/amp-sticky-ad.css [hunk 1]
```diff
@@ -23,6 +23,7 @@ amp-sticky-ad {
   z-index: 2;
   /* TODO(zhouyx, #3250): discuss on what the max-height should be? */
   max-height: 100px !important;
+  margin-bottom: 0 !important;
```
Comment (jridgewell) [inline extensions/amp-sticky-ad/0.1/amp-sticky-ad.css]: The same effect can be done with a transparent background color and `padding-bottom`...

Comment (zhouyx) [inline extensions/amp-sticky-ad/0.1/amp-sticky-ad.css]: I didn't quite get your point here. I don't want a transparent background. And I don't see how `padding-bottom` would achieve the same effect as `margin-bottom` Could you explain it a bit?

Comment (jridgewell) [inline extensions/amp-sticky-ad/0.1/amp-sticky-ad.css]: > I don't want a transparent background.

`background` can be user specified, so can `padding-bottom`.

Try adding both of those to the `<amp-sticky-ad>` examples, and you'll get the same effect as adding a `margin-bottom`.

- - -

We'll also need to add this to the `amp-sticky-ad > amp-ad` selector.

Comment (zhouyx) [inline extensions/amp-sticky-ad/0.1/amp-sticky-ad.css]: I see your point! Yes I should also add `padding-bottom: 0 !important`. But for `background` I don't know, I mean it should be customizable and I don't see why we shouldn't allow user to make sticky-ad background-color to the match the style of its parent.

Comment (jridgewell) [inline extensions/amp-sticky-ad/0.1/amp-sticky-ad.css]: I'm not saying we should mandate a `background`, just pointing out that `padding-bottom` can be used for the same effect with a transparent background. We need to fix the `padding-bottom` issue.

File: extensions/amp-sticky-ad/0.1/amp-sticky-ad.css [hunk 2]
```diff
@@ -23,6 +23,11 @@ amp-sticky-ad {
...
+  margin-left: 0 !important;
```
Comment (zhouyx) [inline extensions/amp-sticky-ad/0.1/amp-sticky-ad.css]: I add more restrictions that should be apply to the style. PTAL @jridgewell

**Comment by lannka** [comment]
Sorry I joined late to the conversation. But honestly I feel this is something AMP shouldn't worry about?

Is it really our policy to forbid any padding/border/margin?

File: extensions/amp-sticky-ad/0.1/amp-sticky-ad.css [hunk 3]
```diff
@@ -24,6 +24,10 @@ amp-sticky-ad {
   /* TODO(zhouyx, #3250): discuss on what the max-height should be? */
   max-height: 100px !important;
   margin-bottom: 0 !important;
+  margin-left: 0 !important;
```
Comment (jridgewell) [inline extensions/amp-sticky-ad/0.1/amp-sticky-ad.css]: Left and right shouldn't be necessary, and may actually be required by the user when showing a 320w ad on a larger phone.

File: extensions/amp-sticky-ad/0.1/amp-sticky-ad.css [hunk 4]
```diff
@@ -24,6 +24,10 @@ amp-sticky-ad {
...
+  margin-right: 0 !important;
+  padding-top: 0 !important;
```
Comment (jridgewell) [inline extensions/amp-sticky-ad/0.1/amp-sticky-ad.css]: I don't think top should be restricted either.

**Comment by zhouyx** [comment]
@lannka I think this is the special case for sticky-ad, where we don't want it to float, but sticky to the very bottom of the page.

File: extensions/amp-sticky-ad/0.1/amp-sticky-ad.css (hunk 3 above)
Comment (zhouyx) [inline extensions/amp-sticky-ad/0.1/amp-sticky-ad.css]: This is for `amp-sticky-ad` which will always has `width:100%` with `margin-left` for example close button could be hidden.

File: extensions/amp-sticky-ad/0.1/amp-sticky-ad.css (hunk 4 above)
Comment (zhouyx) [inline extensions/amp-sticky-ad/0.1/amp-sticky-ad.css]: then it will also create a gap between sticky-ad top and the amp-ad top?

**Comment by zhouyx** [comment]
Not sure how restrictive we should be, as @lannka points our user could add padding/margin on purpose to have their own footer on bottom of the page. 
let's defer this PR.

**Pull request closed without merging**